lib/Importer/Caffe2ModelLoader.cpp

```diff
@@ -464,6 +464,13 @@ void Caffe2ModelLoader::loadOperator(const caffe2::OperatorDef &op) {
     return;
   }
 
+  if (typeName == "LengthsToRanges") {
```

Comment:
In the ONNX loader it would be great to make all these
```
assert(getLengths().dims().size() == 1 && "Lengths must be 1D vector");
  assert(getResult().dims().size() == 2 && "Ranges must be 2D vector");
  assert(getResult().dims()[0] == getLengths().dims()[0] &&
         "Lengths and Ranges must have the same outer dimensions");
  assert(getResult().dims()[1] == 2 && "Inner dimension of Ranges must be 2");
```
checks in the loader and return false if anything fails.

Reply:
Interesting point here.  All the verification stuff is done very late in the game, which is really problematic for ONNX.

Reply:
I do not think we'd need to care for C2 early stage verifications (not going to be used in prod anyway).
It's just ONNXModelLoader needs to handle this properly early on.